client/product_details/ProductDetails.jsx

```diff
@@ -11,9 +11,44 @@ import ProductOverview from './ProductOverview.jsx';
 class ProductDetails extends React.Component {
   constructor(props) {
     super(props);
+    
+    fetch(`http://localhost:8080/products/${this.props.product_id}`)
+      .then((response) => {
+        return response.json();
+      }) 
+      .then((data) => {
+        this.setState({
```

Comment:
You can move this into the above .then() block right after response.json() (which is synchronous) to cut out this .then()